V1 Done and working

Trubador opened the pull request.

Trubador committed:
V1 done

Trubador committed:
Lots of improvements

Trubador committed:
clean up

Trubador committed:
removed password secret

Trubador committed:
Create main.yml

Trubador committed:
Create Dockerfile

Trubador commented:
Any update to my Pull Request? - is it obsolete by now?

Trubador committed:
Fix foreign key constraint on FlowRun table

Trubador committed:
Merge pull request #1 from Trubador/fix-foreign-key-constraint

Fix foreign key constraint on FlowRun table

Trubador committed:
Normalize line endings

Trubador committed:
Save local changes before pulling

Trubador committed:
fixes

Trubador committed:
fix

Trubador committed:
small fix

Trubador committed:
Fix foreign key constraint issue in FlowRun table



---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/Trubador/didact-engine?shareId=XXXX-XXXX-XXXX-XXXX).

Trubador committed:
Merge pull request #2 from Trubador/fix-foreign-key-constraint

Fix foreign key constraint issue in FlowRun table

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/DidactHQ/didact-engine?pullRequest=5) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/DidactHQ/didact-engine?pullRequest=5) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/DidactHQ/didact-engine?pullRequest=5) it.</sub>

Trubador committed:
Add ExecutionMode support and refactor database initialization

- Introduced ExecutionMode entity and updated related configurations.
- Added FlowRunRepository for managing flow runs.
- Updated database context to include ExecutionModes DbSet.
- Refactored database initialization logic to seed ExecutionModes.
- Updated application to use SQLite as the default database provider.
- Enhanced error handling and logging in various components.
